etna/transforms/math/differencing.py

```diff
@@ -124,11 +130,8 @@ def transform(self, df: pd.DataFrame) -> pd.DataFrame:
         transformed = df.loc[:, pd.IndexSlice[segments, self.in_column]].copy()
         for current_segment in segments:
             to_transform = transformed.loc[:, pd.IndexSlice[current_segment, self.in_column]]
-            start_idx = to_transform.first_valid_index()
             # make a differentiation
-            transformed.loc[start_idx:, pd.IndexSlice[current_segment, self.in_column]] = to_transform.loc[
-                start_idx:
-            ].diff(periods=self.period)
+            transformed.loc[:, pd.IndexSlice[current_segment, self.in_column]] = to_transform.diff(periods=self.period)
```

Comment:
May be be should rework the current test to check the new behaviour, I mean run tests in fixtures with NaNs

Reply:
What tests do you mean?

Reply:
Current tests for `DifferencingTransform` (`test_differencing_transform`) have a fixture with NaNs at the beginning. And tests check that everything is working.

Reply:
Do we want to add more tests in `test_differencing_transform` or tests in `test_inference` are enough?
Ok, I'll look at it.